modules/scheduler/gke-cluster/main.tf

```diff
@@ -168,6 +176,14 @@ resource "google_container_cluster" "gke_cluster" {
     ignore_changes = [
       node_config
     ]
+    precondition {
+      condition     = !(!coalesce(var.enable_dataplane_v2, true) && (coalesce(var.enable_multi_networking, false) || length(var.additional_networks) > 0))
```

Comment:
nit: I would recommend the following

```
!(var.enable_dataplane_v2 == false && local.derived_enable_multi_networking)
```

The two changes being 1. direct comparison to false and 2. using the local value.

I know we agreed not to use local, but in this case I think we are just duplicating logic that you have already encoded into `local.derived_enable_multi_networking`. Furthermore, now that your logic for `local.derived_enable_multi_networking` always respects user inputs I think the case for using it is more clear cut. Furthermore, this pre-condition is really about validating `var.enable_dataplane_v2` which we are still using direct reference to input from customer.

---
**Not NIT**

An independent concern is that you currently have something like

`A & B | C` which evaluates to equivalent of `(A & B) | C`
This is not equivalent to `A & (B | C)` which I believe is what you are really after. This is a moot point if you make the change recommended above.

If you chose to stick with the current format I think you need an additional set of parens:

```
!(!coalesce(var.enable_dataplane_v2, true) && ((coalesce(var.enable_multi_networking, false) || length(var.additional_networks) > 0)))
```

Reply:
Updated to:
`!(!coalesce(var.enable_dataplane_v2, true) && local.derived_enable_multi_networking)`

This code `var.enable_dataplane_v2 == false` throws error if it is null.

Reply:
About "Not NIT" part, Please correct me if I'm wrong. I think the additional parens were not needed because the suggestion would change `!(!A && (B | C))` to `!(!A && ((B | C)))` which do not seem different.

Reply:
Looking again, maybe I missed a parens.